src/clinvar_combiner/core.clj

```diff
@@ -163,78 +163,85 @@
-                                  ;"clinical_assertion_trait_set_clinical_assertion_trait_ids"
-                                  "clinical_assertion_trait"
-                                  "gene"
-                                  "variation"
-                                  "gene_association"
-                                  "variation_archive"
-                                  "rcv_accession"
-                                  "clinical_assertion"
-                                  "clinical_assertion_observation"
-                                  ;"clinical_assertion_observation_ids"
-                                  "clinical_assertion_variation"
-                                  ;"clinical_assertion_variation_descendant_ids"
-                                  "trait_mapping"
-                                  ]]
-                      (doseq [table-name tables-to-clean]
-                        (let [updated-count (jdbc/execute! @db-client/db [(format "update %s set dirty = 0 where dirty = 1" table-name)])]
-
-                          (log/infof "Marked %s records in table %s as clean" updated-count table-name))))
-                    )
-                  )                                         ; end if end sentinel
+              (let [release-sentinel (json/parse-string (second kv) true)
+                    sentinel-type (get-in release-sentinel [:content :sentinel_type])]
+                (cond (= "start" sentinel-type)
+                      (do (let [record-json (json/generate-string release-sentinel)]
+                            (log/info "Got start sentinel" record-json)
+                            (jc/produce! producer (:output topic-metadata) record-json)))
+
+                      (= "end" sentinel-type)
+                      ; Flush non-SCVs
+                      ; records-to-flush is lazy, avoid realizing it
+                      (do
+                        (doseq [record (sink/get-dirty release-sentinel)]
+                          (let [record-json (json/generate-string record)]
+                            (log/info record-json)
+                            (jc/produce! producer (:output topic-metadata) record-json)))
+
+
+                        ; Flush SCVs
+                        (let [scvs-to-flush (sink/dirty-bubble-scv release-sentinel)]
+                          (log/info release-sentinel)
+                          (log/infof "Received %d messages to flush" (count scvs-to-flush))
+                          (doseq [clinical-assertion scvs-to-flush]
+                            (let [built-clinical-assertion (sink/build-clinical-assertion clinical-assertion)
+                                  built-clinical-assertion (sink/post-process-built-clinical-assertion built-clinical-assertion)
+                                  built-clinical-assertion-json (json/generate-string built-clinical-assertion)]
+                              (assert (not (nil? (:id built-clinical-assertion)))
```

Comment:
Assertions vs Exceptions?

Reply:
It will throw a `java.lang.AssertionError` if the condition fails but maybe it's better to have a more explicit exception with the cause provided, and to match all the other error handling that uses `ex-info`. I'll update this